paparazzi-annotations/README.md

```diff
@@ -0,0 +1,114 @@
+# `@Paparazzi`
+An annotation used to generate Paparazzi snapshots for composable preview functions.
+
+## Installation
+Add the following to your `build.gradle` file
+
+```groovy
+apply plugin: 'app.cash.paparazzi.preview'
+```
+
+## Basic Usage
+Apply the annotation alongside an existing preview method. The annotation processor will generate a manifest of information about this method and the previews applied.
+
+```kotlin
+import app.cash.paparazzi.preview.Paparazzi
+
+@Paparazzi
+@Preview
+@Composable
+fun MyViewPreview() {
+  MyView(title = "Hello, Paparazzi Annotation")
+}
+```
+
+Run `:recordPaparazziDebug` in your module to generate preview snapshots (and optionally verify them using `:verifyPaparazziDebug`) as you normally would.
+
+A test class to generate snapshots for annotated previews will automatically be generated.
+If you prefer to define a custom snapshot test, you mey disable test generation by adding the following to your `build.gradle` file.
+
+```groovy
+paparazziPreview {
+  generateTestClass = false
```

Comment:
What's the proposed use case where someone would mark a `@Preview` with `@Paparazzi`, but not want a test class to be generated as a result of the latter?

Reply:
you would do this if you wanted to create your own implementation of the generated test class instead of using the codegen version.

Reply:
I understand that from the property naming convention, but can you elaborate more on a use case where this might be desirable?  And how would that differ from today where one manually crafts the test class?

Unless I'm missing something, a nice layered design decision would be to either write the test manually or let the annotation generate it for you.  Otherwise, integrating with customizations and keeping up with Google's @Preview and @PreviewParameter exposes us to maintenance burden.

Reply:
I ended up making this a gradle property instead of gradle extension dsl.

Reply:
I see that, however, that still bleeds the implementation detail of `PaparazziPreviewData`, per

> Especially, given the plan to replace this test source codegen with direct SDK snapshotting anyway, I'd be even more wary of exposing customization hooks from the get-go.